Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift

```diff
@@ -1193,6 +1210,23 @@ public struct RenderNodeTranslator: SemanticVisitor {
                 .filter({ !($0.unconditionallyUnavailable == true) })
                 .sorted(by: AvailabilityRenderOrder.compare)
         )
+
+        if let availability = documentationNode.metadata?.availability, !availability.isEmpty {
+            let renderAvailability = availability.compactMap({
+                let currentPlatform: PlatformVersion?
+                if let name = PlatformName(metadataPlatform: $0.platform),
+                   let contextPlatform = context.externalMetadata.currentPlatforms?[name.displayName] {
+                    currentPlatform = contextPlatform
+                } else {
+                    currentPlatform = nil
+                }
```

Comment:
Minor style preference: Since the `else` body sets `currentPlatform` to `nil` this same thing could be expressed with an [optional flatMap][1] (this also applied above).

```suggestion
                let currentPlatform = PlatformName(metadataPlatform: $0.platform).flatMap { name in
                    context.externalMetadata.currentPlatforms?[name.displayName]
                }
```

[1]: https://developer.apple.com/documentation/swift/optional/flatmap(_:)